hotshot-state-prover/src/circuit.rs

```diff
@@ -0,0 +1,541 @@
+    ///  `vars[1]`: block height
+    ///  `vars[2]`: block commitment root
+    ///  `vars[3]`: fee ledger commitment
+    ///  `vars[4-6]`: stake table commitment
+    vars: [Variable; 7],
+}
+
+impl LightClientStateVar {
+    /// # Errors
+    /// if unable to create any of the public variables
+    pub fn new<F: PrimeField>(
+        circuit: &mut PlonkCircuit<F>,
+        state: &GenericLightClientState<F>,
+    ) -> Result<Self, CircuitError> {
+        let view_number_f = F::from(state.view_number as u64);
+        let block_height_f = F::from(state.block_height as u64);
+        Ok(Self {
+            vars: [
+                circuit.create_public_variable(view_number_f)?,
+                circuit.create_public_variable(block_height_f)?,
+                circuit.create_public_variable(state.block_comm_root)?,
+                circuit.create_public_variable(state.fee_ledger_comm)?,
+                circuit.create_public_variable(state.stake_table_comm.0)?,
+                circuit.create_public_variable(state.stake_table_comm.1)?,
+                circuit.create_public_variable(state.stake_table_comm.2)?,
+            ],
+        })
+    }
+
+    /// Returns the view number
+    #[must_use]
+    pub fn view_number(&self) -> Variable {
+        self.vars[0]
+    }
+
+    /// Returns the block height
+    #[must_use]
+    pub fn block_height(&self) -> Variable {
+        self.vars[1]
+    }
+
+    /// Returns the Merkle root of the block commitments
+    #[must_use]
+    pub fn block_comm_root(&self) -> Variable {
+        self.vars[2]
+    }
+
+    /// Returns the commitment of the fee ledger
+    #[must_use]
+    pub fn fee_ledger_comm(&self) -> Variable {
+        self.vars[3]
+    }
+
+    /// Returns the commitment of the associated stake table
+    #[must_use]
+    pub fn stake_table_comm(&self) -> StakeTableCommVar {
+        StakeTableCommVar {
+            qc_keys_comm: self.vars[4],
+            state_keys_comm: self.vars[5],
+            stake_amount_comm: self.vars[6],
+        }
+    }
+}
+
+impl AsRef<[Variable]> for LightClientStateVar {
+    fn as_ref(&self) -> &[Variable] {
+        &self.vars
+    }
+}
+
+/// A function that takes as input:
+/// - a list of stake table entries (`Vec<(SchnorrVerKey, Amount)>`)
+/// - a bit vector indicates the signers
+/// - a list of schnorr signatures of the updated states (`Vec<SchnorrSignature>`), default if the node doesn't sign the state
+/// - updated light client state (`(view_number, block_height, block_comm_root, fee_ledger_comm, stake_table_comm)`)
+/// - a quorum threshold
+/// Lengths of input vectors should not exceed the `STAKE_TABLE_CAPACITY`.
+/// The list of stake table entries, bit indicators and signatures will be padded to the `STAKE_TABLE_CAPACITY`.
+/// It checks that
+/// - the signer's accumulated weight exceeds the quorum threshold
+/// - the stake table corresponds to the one committed in the light client state
+/// - all signed Schnorr signatures are valid
```

Comment:
Also that:  "The vector that indicates who signed is a bit vector."

Reply:
i think it's implicit that bit vector would be constrained to be a boolean vector.
don't think we need to include as part of the circuit description here.

Reply:
As mentioned [below](https://github.com/EspressoSystems/espresso-sequencer/pull/1041#discussion_r1489942751), the current implementation is correct, yet ensuring the vector is composed by booleans is a core requirement of the circuit that is worth mentioning I think.

Reply:
Done in [24bb979](https://github.com/EspressoSystems/espresso-sequencer/pull/1041/commits/24bb97974fd20657011734e458dd589788ed1765)

Reply:
Thanks!